src/main/kotlin/no/nav/syfo/dialogmote/domain/ArbeidstakerBrev.kt

```diff
@@ -0,0 +1,12 @@
+package no.nav.syfo.dialogmote.domain
+
+import java.time.LocalDateTime
+import java.util.UUID
+
+abstract class ArbeidstakerBrev {
+    abstract val uuid: UUID
+    abstract val motedeltakerArbeidstakerId: Int
+    abstract val document: List<DocumentComponentDTO>
+    abstract val pdf: ByteArray
+    abstract val lestDatoArbeidstaker: LocalDateTime?
+}
```

Comment:
Skjønner hva du prøver på her, men går det ann å få til en refaktorering uten bruk av arv her? Føler

Synes det er en fordel å unngå arv så langt det er hensiktmessig.

Reply:
Jeg synes arv fungerer bra i dette tilfellet (se på kodeendingene i ArbeidstakerVarselApi) - hva synes du er problematisk med det?

Det går an å gjøre om ArbeidstakerBrev til en konkret klasse, og la Referat (og DialogmotedeltakerArbeidstakerVarsel) ha en sånn som en property. Men det som blir vrient da for Referat er et mye av det samme innholdet også gjelder for *arbeidsgiver*, og da blir det litt rart at det ligger i en arbeidstakerBrev-instans. Det å duplisere innholdet blir også probelmatisk synes jeg.

Reply:
Ser den!
Er ikke krise, men føler ting begynner å bli litt uoversiktlig. Det er kanskje eneste ålreit måte å løse det på så lenge det underliggende dataene er som de er.

Er vel et symptom på at det som arbeidsaker leser burde vært skilt fra dokumentene(innkalling, endring, avlysning, referat). Er kanskje litt sent å begynne å rydde opp i det nå :/

Modellen burde kanskje egentlig vært mer av typen.
- Referat(dokumentet),
- Innkalling(dokumentet)
- EndringTidSted(dokumentet),
- Avlysning(dokumentet),

- Varsel/brev(det som leses)

Reply:
Det kan godt være at det går an å forbedre modellen, men jeg synes det fungerer ganske bra slik den er nå.

Det blir ekstra kompleksitet av at det er *en instans* av referat for et møte, mens for de andre hendelsene er det egne objekter for hhv arbeidstaker og arbeidsgiver. Men dette reflekterer jo en reell forskjell, så jeg synes det er greit.

Jeg har sett litt mer på forslaget og det er mulig å erstatte den superklassen jeg lagde med et interface. Kanskje det er bedre?

Reply:
Høres fint ut det 👍